adagucserverEC/adagucserver.cpp

```diff
@@ -204,15 +290,35 @@ int _main(int argc, char **argv, char **) {
     CDBError("And --tailpath for scanning specific sub directory, specify --path for a absolute path to update");
     return 1;
   } else if (((scanFlags & CDBFILESCANNER_UPDATEDB) == CDBFILESCANNER_UPDATEDB) && (configSet == 1)) { /* Update database */
-    CRequest request;
-    status = setCRequestConfigFromEnvironment(&request);
-    if (status != 0) {
-      CDBError("Unable to read configuration file");
+
+    if (!layerPathToScan.empty() && !CDirReader::isFile(layerPathToScan.c_str())) {
+      CDBError("Provided file does not exist [%s].", layerPathToScan.c_str());
       return 1;
     }
-    status = request.updatedb(&tailPath, &layerPathToScan, scanFlags, layerName);
-    if (status != 0) {
-      CDBError("Error occured in updating the database");
+    std::set<std::string> datasetsToScan;
+
+    if (automaticallyFindMatchingDataset) {
+      datasetsToScan = findDataSetsToScan(layerPathToScan, verbose);
+      CDBDebug("Datasets to scan: %d", datasetsToScan.size());
+    } else {
+      // Empty string to make for loop work.
+      datasetsToScan.insert("");
```

Comment:
This pushes an empty string to the set.
Is an empty string as dataset name handled correctly  in the for loop  (lines 307-321)?

Reply:
Yes that will work. Set `CRequestConfigFromEnvironment(CRequest *request, CT::string dataset = "")`; will work just as before when the dataset is empty.

Reply:
OK